src/interface/storage.thrift

```diff
@@ -61,10 +66,22 @@ struct ResultCode {
     3: optional common.HostAddr  leader,
 }
 
+struct EdgeData {
+    1: common.EdgeType type,
+    2: common.Schema   schema,
+    3: binary          data,   // decode according to edge_schema.
+}
+
+struct TagData {
+    1: common.TagID          tag_id,
+    2: common.Schema         schema,
+    3: binary                vertex_data,
+}
+
 struct VertexData {
-    1: common.VertexID vertex_id,
-    2: binary vertex_data, // decode according to vertex_schema.
-    3: binary edge_data,   // decode according to edge_schema.
+    1: common.VertexID       vertex_id,
+    2: list<TagData>         tag_data,
```

Comment:
Because it is possible that a VertexID belongs to a different tag.

Reply:
Currently,  if user want to fetch different props on different tags,   storage will return them in one line.  For example,   tag1.prop1,  tag2.prop1,  tag2.prop2,  tag3.prop1 ....

And we encode them together,  so query engine could use it directly. 

Of course,  for multi edges,  there is the same problem.  For example,  what's the output for this go sentence now :  "go from id over edge1, edge2 return edge1.prop1, edge2.prop1".   Do we support it now?

Reply:
Currently,  if user want to fetch different props on different tags,   storage will return them in one line.  For example,   tag1.prop1,  tag2.prop1,  tag2.prop2,  tag3.prop1 ....

And we encode them together,  so query engine could use it directly. 

Of course,  for multi edges,  there is the same problem.  For example,  what's the output for this go sentence now :  "go from id over edge1, edge2 return edge1.prop1, edge2.prop1".   Do we support it now?

Reply:
Yes, we can handle this.

Reply:
If so,  why not encode them in one line.  use one schema is enough

Reply:
That's because different tag have different schema.

Reply:
This is result set.  It has no relationship with tag schema.

Reply:
When returning the same prop name for multiple tags, there is no good way for the server to get the type of prop returned, so we need "list<TagData>".

Reply:
Not exactly,  we have fixed the problem.  Query engine will use index to get the related tag prop,  and storage return the props as the order passed in.

Reply:
This will cause a problem, assuming you will take tag1.prop1, tag2.prop2, tag1.prop2. If tag2.prop2 does not exist, then the graph server will not know if tag2.prop2 does not exist or tag1.prop2 does not exist.

Reply:
We have talked about this problem.  It should be fixed by null after it is supported in DataMan.

Reply:
It can fix this problem with each tag_data with its own schema, or modify the struct ColumnDef and adding tagid/edgetype to it.

Reply:
Not only this.  For the whole response,  we only have two schema,  the tag result and the edge result.  We have no needs to send schema in each vertex data. It is too cost.

Reply:
I know what you mean. If this is the case, I think it would be better to modify the struct ColumnDef(add EdgeType/TagID).

Reply:
Because if we use index, then the name field in ColumnDef is meaningless.